DISA STIG refactoring

shawndwells opened the pull request:
- Renamed stig-rhel7-server-upstream to stig-rhel7-upstream, and incorporated the misc GUI checks
- Net-new profile stig-rhel7-disa that aligns to only controls released in official DISA STIG (as best we can keep to their intent, without introducing the errors from their content)

shawndwells commented:
@redhatrises refactoring the DISA STIG profile setup a bit, that included your work. not done yet -- need to incorporate into the build system -- but what do you think?

shawndwells commented:
The misc TO DO items from https://github.com/OpenSCAP/scap-security-guide/pull/1773/files#diff-45d6c53ee1f65af0fc3906c889693da3 reflect DISA provided rules that SSG doesn't have a mapping to. Likely just a mapping exercise vs needing new rules.

redhatrises commented:
@shawndwells looks like the build system is still looking for the old profiles that you removed which is why the build is failing. Check the XSLT in `RHEL/7/transforms` as well as in `cmake/SSGCommon.cmake`.

Also there should be a separate profile for STIGs with GUIs and STIGs without GUIs.

shawndwells commented:
I don't see references to old profiles:

``
$ grep -rin "server-upstream"
Binary file .git/index matches
RHEL/5/input/guide.xslt:19:       <xsl:apply-templates select="document('profiles/stig-rhel5-server-upstream.xml')" />
RHEL/5/input/profiles/stig-rhel5-server-upstream.xml:1:<Profile id="stig-rhel5-server-upstream">
RHEL/6/input/auxiliary/DISCLAIMER:1:The upstream STIG for Red Hat Enterprise Linux 6 Server profile "stig-rhel6-server-upstream" is
RHEL/6/input/guide.xslt:27:       <xsl:apply-templates select="document('profiles/stig-rhel6-server-upstream.xml')" />
RHEL/6/input/profiles/stig-rhel6-server-gui-upstream.xml:1:<Profile id="stig-rhel6-server-gui-upstream" extends="stig-rhel6-server-upstream">
RHEL/6/input/profiles/stig-rhel6-server-upstream.xml:1:<Profile id="stig-rhel6-server-upstream" extends="common">
RHEL/6/kickstart/ssg-rhel6-stig-ks.cfg:376:oscap xccdf eval --remediate --profile stig-rhel6-server-upstream --report /root/oscap_stig_remediation_report.html \
RHEL/6/CMakeLists.txt:18:ssg_build_html_stig_tables(${PRODUCT} "stig-${PRODUCT}-server-upstream" ${DISA_STIG_VERSION})
SUSE/11/input/auxiliary/DISCLAIMER:1:The upstream STIG for Red Hat Enterprise Linux 6 Server profile "stig-rhel6-server-upstream" is
SUSE/12/input/auxiliary/DISCLAIMER:1:The upstream STIG for Red Hat Enterprise Linux 6 Server profile "stig-rhel6-server-upstream" is
docs/SCAP_and_STIG_Workshop/en-US/Content_Customization.xml:272:-rw-rw-r--. 1 dave dave  4736 Mar  5 12:10 stig-rhel6-server-upstream.xml
docs/scap-security-guide.8:131:.I stig-rhel6-server-upstream
docs/scap-security-guide.8:295:.I stig-rhel7-server-upstream
docs/scap-security-guide.8:366:stig-rhel6-server-upstream profile:
docs/scap-security-guide.8:368:oscap  xccdf eval --profile stig-rhel6-server-upstream \
scap-security-guide.spec.in:138:+ RHEL6 stig-rhel6-server XCCDF profile renamed to stig-rhel6-server-upstream
``

shawndwells commented:
CMake appears to have the new profiles...

`````
$ grep -rin "}\-disa" *
RHEL/7/CMakeLists.txt:23:ssg_build_html_stig_tables(${PRODUCT} "stig-${PRODUCT}-disa" ${DISA_STIG_VERSION})
`````

shawndwells commented:
ok, factored the commits. 

- Added DISA STIG profile that maps to only the rules selected by DISA (rhel7-stig-disa)
- Combined the GUI rules into the core stig-upstream profile, to align (for now) with how DISA does things
- Dropped original commits that put the new stig-rhel7-disa profile into the build system. Couldn't figure out cmake and wanted to get something committed to build collaboration base for the new profile

shawndwells commented:
(ready for review to get the base profile in. it's not linked to build system yet. will allow us to start as a community to add missing rules to the stig-rhel7-disa skeleton)

redhatrises commented on RHEL/7/input/profiles/stig-rhel7-disa.xml:
```diff
@@ -0,0 +1,752 @@
+<Profile id="stig-rhel7-disa">
+<title>STIG for Red Hat Enterprise Linux 7</title>
+<description>
+This profile aligns with the DISA STIG for Red Hat Enterprise Linux V1R1, released by DISA on xx-MARCH-2017.
```
@shawndwells should we really have STIG version and release date in this profile? Seems like something else that would have to manually be updated which could be annoying.

shawndwells replied:
done

shawndwells commented:
> On Mar 31, 2017, at 5:41 AM, redhatrises <notifications@github.com> wrote:
> 
> @redhatrises commented on this pull request.
> 
> In RHEL/7/input/profiles/stig-rhel7-disa.xml:
> 
> > @@ -0,0 +1,752 @@
> +<Profile id="stig-rhel7-disa">
> +<title>STIG for Red Hat Enterprise Linux 7</title>
> +<description>
> +This profile aligns with the DISA STIG for Red Hat Enterprise Linux V1R1, released by DISA on xx-MARCH-2017.
> @shawndwells should we really have STIG version and release date in this profile? Seems like something else that would have to manually be updated which could be annoying.
> 

Good point. Calling out the DISA version is important, but the date might Elise that SSG is the one being tardy on releases . Will drop it shortly .

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

shawndwells commented:
note the profile hasn't been incorporated into the build system. seeking to merge now to give a common base to build from as new rules are created to align with DISA content.

redhatrises commented:
@shawndwells I understand that this is to align with DISA content, but I am slightly concerned about profiles that have fully GUI checks for systems that do not run GUIs. Not sure if we can have an unofficial version that doesn't incorporate GUI checks? I think that the user community would be more appreciative of that.

shawndwells commented:
@redhatrises we lost the battle with DISA. On the plus side, this means the RHEL7 STIG works for RHEL7 Server, Workstation, Client, Desktop, HPC, and Atomic. *If* the checks verify X/GNOME is installed before running, would this be a burden to users?

redhatrises commented:
> @redhatrises we lost the battle with DISA. On the plus side, this means the RHEL7 STIG works for RHEL7 Server, Workstation, Client, Desktop, HPC, and Atomic. If the checks verify X/GNOME is installed before running, would this be a burden to users?

Thanks. Good point and probably not.

Ack.

shawndwells commented:
Thanks @redhatrises !
